featurebyte/session/base_spark.py

```diff
@@ -77,11 +82,30 @@ def _initialize_storage(self) -> None:
         if self.storage_type == StorageType.FILE:
             self._storage = FileSimpleStorage(storage_url=self.storage_url)
         elif self.storage_type == StorageType.S3:
+            if self.storage_credential is None:
```

Comment:
can consider extracting to a function for reuse with the GCS validation

eg. `def _validate_storage_credential(credential, expected_type)`

Reply:
Will use typecheck instead to simplify credential checking

Reply:
If we move the type checking elsewhere mypy type checking fails. Will keep current implementation first and optimize later when adding other storage types